packages/storage_client/lib/src/types.dart

```diff
@@ -191,6 +194,19 @@ class SignedUrl {
   }
 }
 
+class SignedUploadURLResponse extends SignedUrl {
+  final String token;
+
+  const SignedUploadURLResponse({
+    required String signedUrl,
+    required String path,
+    required this.token,
+  }) : super(
+          signedUrl: signedUrl,
+          path: path,
+        );
```

Comment:
Can't we use the new super constructors?

```suggestion
  const SignedUploadURLResponse({
    required super.signedUrl,
    required super.path,
    required this.token,
  });
```

Reply:
That's a great point. Currently, the minimum Dart SDK version is 2.15.0, and I believe the new super constructor was introduced on v2.17.0, so let's actually do that in another PR. I think we can bump the minimum SDK version to 2.17.0 and minimum Flutter version to 3.0.0 soon, since it's been a while for those to be around.